### PR TITLE
[WIP] dynamodb_table: boto3, AnsibleAwsModule, billing_mode, point_in_time_recovery and more

### PR DESCRIPTION
**Pull request by marknet15**
### SUMMARY
This is a fresh PR to transfer over the original PR from the ansible repo to the new community.aws repo as requested to meet new requirements.

#### Original PR's
https://github.com/ansible/ansible/pull/45402
https://github.com/ansible/ansible/pull/68320
https://github.com/anryko/ansible/pull/1

I'm providing the fixes listed below for the changes proposed in https://github.com/ansible/ansible/pull/68320 that are now stale that is in turn based on the below:
> Fix AnsibleAwsModule migration stale PR https://github.com/ansible/ansible/pull/45402. Add billing_mode and point_in_time_recovery. > Small bug fixes. Integration tests fixes.

**Fixes:**
* Fix issue with `ProvisionedThroughput` params being set incorrectly on GSI's when billing is `PAY_PER_REQUEST`.
* Fix incorrect `key_type_mapping`.
* Fix defining of `serialized_index_attribute_definitions` for both the hash and the range key on a GSI. (range key definition was being overwritten).
* Fix defining of SSE spec.

Fixes #45402
Fixes #56923

#### ISSUE TYPE
- Bugfix Pull Request
- Feature Pull Request

#### COMPONENT NAME
- dynamodb_table

#### ADDITIONAL INFORMATION
```yaml
- name: enable on-demand billing and backups
  dynamodb_table:
  name: table_name
  hash_key_name: myhashkey
  range_key_name: myrangekey
  billing_mode: PAY_PER_REQUEST
  point_in_time_recovery: yes
  indexes:
      - name: myindex
          type: global_all
          hash_key_name: somekey
          range_key_name: myrangekey
```

### CODE REVIEW

File: tests/integration/targets/dynamodb_table/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+            hash_key_name: myhashkey
+          - name: NamedIndexV2
+            type: global_all
+            hash_key_name: myhashkey
+            range_key_name: myrangekey
+            read_capacity: 10
+            write_capacity: 10
+      register: result
+      until:
+        - result | json_query('indexes[?name==`NamedIndexV2`].read_capacity') == [10]
```
Comment (jillr) [inline tests/integration/targets/dynamodb_table/tasks/main.yml:109]: The `json_query` filter has moved to a collection, so these will need to be `community.general.json_query`.

File: tests/integration/targets/dynamodb_table/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+        - result | json_query('global_secondary_indexes[?IndexName==`NamedIndexV2`].IndexStatus') == ['ACTIVE']
+        - result | json_query('global_secondary_indexes[?IndexName==`NamedIndex`].IndexStatus') == ['ACTIVE']
+      retries: 6
+      delay: 10
+
+    - name: update index on the existing table
+      dynamodb_table:
+        name: "{{ table_name }}"
+        hash_key_name: myhashkey
+        indexes:
```
Comment (jillr) [inline tests/integration/targets/dynamodb_table/tasks/main.yml:97]: AWS permits only one index to be created in a single action, this task is throwing an unhandled ClientError.  The code should reflect and enforce the service limits.
`botocore.errorfactory.LimitExceededException: An error occurred (LimitExceededException) when calling the UpdateTable operation: Subscriber limit exceeded: Only 1 online index can be created or deleted simultaneously per table`
https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Limits.html#limits-secondary-indexes

I haven't done a proper review of the code, but there doesn't seem to be much exception handling.  We generally like to catch `ClientError` and `BotoCoreError`.

**Comment by falsedlah** [comment]
Hi guys, 
What is the status on this pr? Is there something blocking it?  I need this feature and if there is something I can help with to get it move forward, please let me know.

**Comment by marknet15** [comment]
Hey I completely forgot to pick this back up again, I'll try and get it all up-to-date again and address the feedback on the PR

**Comment by tremble** [comment]
I've started working on a simple boto3 migration (#726) while that won't add the extra features it should be easier to review and much simpler to add features to an existing framework.

**Comment by marknet15** [comment]
@tremble Ah I had completely forgotten about this PR, I'll close it then I'll create a fresh PR with the extra functionality after your boto -> boto3 migration

**Pull request closed without merging**